flow/src/main/java/com/sequenceiq/flow/rotation/handler/RollbackRotationHandler.java

```diff
@@ -33,10 +41,20 @@ protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<Rol
 
     @Override
     protected Selectable doAccept(HandlerEvent<RollbackRotationTriggerEvent> event) {
-        Exception exception = event.getData().getException();
-        SecretRotationStep failedStep = event.getData().getFailedStep();
-        secretRotationService.rollbackRotation(event.getData().getSecretType(), event.getData().getResourceCrn(),
-                event.getData().getExecutionType(), failedStep);
-        return RotationFailedEvent.fromPayload(SecretRotationEvent.ROTATION_FAILED_EVENT.event(), event.getData(), exception, failedStep);
+        RollbackRotationTriggerEvent rollbackEvent = event.getData();
+        String resourceCrn = rollbackEvent.getResourceCrn();
+        RotationFlowExecutionType executionType = rollbackEvent.getExecutionType();
+        SecretType secretType = rollbackEvent.getSecretType();
+        Exception exception = rollbackEvent.getException();
+        SecretRotationStep failedStep = rollbackEvent.getFailedStep();
+        secretRotationUsageProcessor.ifPresent(processor -> processor.rollbackStarted(secretType, resourceCrn, executionType));
+        try {
+            secretRotationService.rollbackRotation(secretType, resourceCrn, executionType, failedStep);
+        } catch (Exception e) {
+            secretRotationUsageProcessor.ifPresent(processor -> processor.rollbackFailed(secretType, resourceCrn, e.getMessage(), executionType));
+            throw e;
+        }
+        secretRotationUsageProcessor.ifPresent(processor -> processor.rollbackFinished(secretType, resourceCrn, executionType));
+        return RotationFailedEvent.fromPayload(SecretRotationEvent.ROTATION_FAILED_EVENT.event(), rollbackEvent, exception, failedStep);
```

Comment:
why don't you move this inside the try? at first blink I had the feeling this would run in case of an exception

Reply:
This should run when no exception happens during rollback because the rotation failed.

Reply:
I know, but the try-catch is misleading this way. if you move this into the try part it's obvious at first blink it would run only if there is no exception, without checking the catch part of the code

Reply:
done